gateway/src/main/java/org/georchestra/gateway/security/preauth/PreauthAuthenticationManager.java

```diff
@@ -106,6 +111,8 @@ public static GeorchestraUser map(Map<String, String> requestHeaders) {
         user.setLastName(lastName);
         user.setOrganization(org);
         user.setRoles(roleNames);
+        user.setOAuth2Provider(provider);
+        user.setOAuth2Uid(providerId);
```

Comment:
I'd add a TODO-comment and maybe create an issue on georchestra/georchestra or here, to stress the fact that the oauth2provider / uid field on the user is misnamed and is also used by external (preauthentication) mechanisms.